### ISSUE TITLE
menu empty

### ISSUE THREAD

**Issue by hannesdatta** [tilburgsciencehub/website-flask#45]
The menu is empty when running the site in Docker using `docker compose build` & `docker compose up -d`.

<img width="1390" alt="image" src="https://github.com/tilburgsciencehub/website-flask/assets/16557730/e4026a7a-314e-4ad6-b075-65a9a4bd878e">

I think this may have to do with the fact that the :5000 port was hardcoded in the code, e.g. https://github.com/tilburgsciencehub/website-flask/blob/db3839dc159e0f47fa2a75c30a65568b51a179b1/functions.py#L131 and https://github.com/tilburgsciencehub/website-flask/blob/db3839dc159e0f47fa2a75c30a65568b51a179b1/html_parser.py#L359.

Docker runs this site via gunicorn on 8000 (wouldn't hard code 8000 either - should be something that dynamically adapts...).

**Comment by thierrylahaije** [comment]
Issue has been fixed, root and base url are now dynamic.

**Comment by hannesdatta** [comment]
Hi @thierrylahaije, this did not solve my initial issue: the menu is still empty.

<img width="1363" alt="image" src="https://github.com/tilburgsciencehub/website-flask/assets/16557730/22655cbe-260c-4ac2-b8a5-8db220a60634">

Please fix.

__How to reproduce__
`docker compose build`
`docker compose up -d`

**Comment by thierrylahaije** [comment]
Hi @hannesdatta,

Sorry, I thought the issue had to do with the URL. I checked what is going wrong, and it seems that since the docker-compose.yml has been added, something is going wrong with the defaultdict part of the collections library. Normally, this should be a basic function from the python base, but for some reason, something is going wrong and I have not been able to find out yet why this is happening. I need to further investigate this.
